trieste/acquisition/optimizer.py

```diff
@@ -161,17 +179,46 @@ def optimize_continuous(space: Box, target_func: AcquisitionFunction) -> TensorT
         def _objective() -> TensorType:
             return -target_func(bijector.forward(variable[:, None, :]))  # [1]
 
-        chosen_points = tf.ones([0, len(space.lower)], dtype=trial_search_space.dtype)  # [0, D]
-        for i in tf.range(num_restarts):  # restart optimization
-            variable.assign(bijector.inverse(initial_points[i : i + 1]))  # [1, D]
-            gpflow.optimizers.Scipy().minimize(_objective, (variable,), **opt_kwargs)
-            chosen_points = tf.concat(
-                [chosen_points, bijector.forward(variable)], axis=0
-            )  # [i+1, D]
-
-        chosen_func_values = target_func(chosen_points[:, None, :])  # [num_restarts, 1]
-
-        return tf.gather(chosen_points, tf.argmax(chosen_func_values))  # [1, D]
+        def _perform_optimization(starting_point: TensorType) -> OptimizeResult:
+            variable.assign(bijector.inverse(starting_point))  # [1, D]
+            return gpflow.optimizers.Scipy().minimize(_objective, (variable,), **opt_kwargs)
+
+        pass_count = 0
```

Comment:
You're really using this like a boolean, so it would probably be clearer to have something like:

```suggestion
        successful_optimization = False
```